### PR TITLE
fix(web): render multi-level subtasks in sidebar task tree

### PR DESCRIPTION
**Pull request by carlosflorencio**
Grandchild tasks existed in the store but never appeared in the sidebar because the renderer only walked two levels. This replaces the flat root/subtask split with a recursive tree so any depth is visible, with collapse and sibling-scoped drag-and-drop at every level.

## Important Changes

- Recursive `TaskTreeNode` / `TaskTreeLevel` renderer replaces the hardcoded 2-level sidebar walk
- Depth-aware row indent (`row-indent.ts`) capped at visual depth 6 to protect narrow sidebars
- Mobile task switcher now maps `parentTaskId` so subtasks nest on mobile for the first time
- Gated `POST /api/v1/_test/tasks` harness route seeds depth ≥ 2 chains for e2e (bypasses service depth guard)

## Validation

- `make -C apps/backend fmt test lint`
- `pnpm --filter @kandev/web typecheck lint test run`
- E2E: `sidebar-multilevel-subtasks.spec.ts` (depth-3 desktop + mid-level collapse), `mobile-sidebar-subtasks.spec.ts` (depth-2 mobile), existing depth-1 sidebar spec

## Possible Improvements

Low risk — production CreateTask still enforces depth ≤ 1; this only affects display of existing deep trees and e2e-seeded data.

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: apps/web/lib/sidebar/apply-view.ts [hunk 1]
```diff
@@ -435,6 +435,26 @@ function applySubtaskOrder(
...
+  let total = 0;
+  const stack = [...rootTasks];
+  while (stack.length > 0) {
+    const task = stack.pop()!;
+    total += 1;
+    const children = subTasksByParentId.get(task.id);
+    if (children) stack.push(...children);
+  }
+  return total;
+}
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/lib/sidebar/apply-view.ts:459]: Fixed in 8a9f3e1 — added a `visited` set to `countGroupTasks` with a cycle regression test.

File: apps/web/components/task/mobile/session-task-switcher-sheet-hooks.ts [hunk 2]
```diff
@@ -108,6 +103,9 @@ function toSheetItem(
   return {
     id: task.id,
     title: task.title,
+    // Carry the parent link so the mobile task switcher nests subtasks the same
+    // way the desktop sidebar does (applyView/TaskSwitcher read parentTaskId).
+    parentTaskId: task.parentTaskId ?? undefined,
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/components/task/mobile/session-task-switcher-sheet-hooks.ts:108]: Fixed — `buildKanbanTaskUpsert` now maps `parentTaskId: task.parent_id ?? undefined` so in-session subtask creation nests immediately on mobile.

[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/components/task/mobile/session-task-switcher-sheet-hooks.ts:108]: Fixed — same `parentTaskId` mapping added to `buildKanbanTaskUpsert`.

File: apps/web/lib/sidebar/apply-view.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/lib/sidebar/apply-view.ts:459]: Fixed — added a `visited` set guard with a cycle regression test in `apply-view-count.test.ts`.

File: apps/web/components/task/task-switcher-subtask-dnd.tsx [hunk 3]
```diff
@@ -39,101 +59,74 @@ function SortableSubtaskRow({ taskId, children }: { taskId: string; children: Re
...
-      {children}
+      <div
+        {...sortableAttributes}
+        {...listeners}
+        // Strip dnd-kit's default tabIndex={0}: only PointerSensor is wired,
+        // so keyboard tab stops here lead nowhere. If KeyboardSensor is
+        // added later, drop this override.
+        tabIndex={undefined}
+        data-testid="sortable-task-handle"
+        className="cursor-grab active:cursor-grabbing"
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/components/task/task-switcher-subtask-dnd.tsx:66]: Fixed — `SortableTaskLevel` now passes `isDraggable` per level; non-draggable nodes render a plain wrapper without `useSortable` or the grab cursor class.

File: apps/web/components/task/task-switcher.tsx [hunk 4]
```diff
@@ -213,54 +205,89 @@ function TaskRow({
...
+  ctx: TaskTreeContext;
+}) {
+  const subs = ctx.subTasksByParentId.get(task.id);
+  const hasSubs = !!subs?.length;
+  const subsHidden = hasSubs && !!ctx.onToggleSubtasks && ctx.collapsedSubs.has(task.id);
+  const toggleInfo: SubtaskToggleInfo | undefined =
+    hasSubs && ctx.onToggleSubtasks
+      ? {
+          subtaskCount: subs!.length,
+          subtasksCollapsed: subsHidden,
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/components/task/task-switcher.tsx:241]: Fixed — subtask toggle badge now uses `countGroupTasks(subs!, ctx.subTasksByParentId)` so the count matches all rows hidden on collapse.

File: apps/web/components/task/task-switcher-subtask-dnd.tsx [hunk 5]
```diff
@@ -21,7 +21,30 @@ import type { TaskSwitcherItem } from "./task-switcher";
...
+/**
+ * A single sortable node in the (arbitrarily deep) task tree.
+ *
+ * `setNodeRef` and the CSS transform live on the outer wrapper so the row and
+ * its nested subtree move together while dragging. Drag listeners attach ONLY
+ * to the `handle` (the task row) — `nested` (the child subtree) renders as a
+ * sibling outside the handle so a pointer-down on a descendant row drags that
+ * descendant, not this node. This is the same handle/children split the root
+ * level has always used; generalizing it is what makes nesting safe at depth.
+ */
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/components/task/task-switcher-subtask-dnd.tsx]: Fixed — moved the architectural JSDoc to the exported `SortableTaskNode` wrapper; `DraggableSortableTaskNode` keeps a one-line note about hook isolation.
